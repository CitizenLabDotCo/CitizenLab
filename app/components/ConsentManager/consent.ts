import { get, set, remove } from 'js-cookie';
import { IConsentCookie } from '.';

const COOKIE_NAME = 'cl2_consent';

export function getConsent() {
  try {
    const json = get(COOKIE_NAME);
    return json ? JSON.parse(json) : null;
  } catch (error) {
    return null;
  }
}

export function setConsent(consent: IConsentCookie) {
  JSON.stringify(consent);
  set(COOKIE_NAME, consent, { expires: 60 });
}

export function removeConsent() {
  remove(COOKIE_NAME, { expires: 60 });
}
