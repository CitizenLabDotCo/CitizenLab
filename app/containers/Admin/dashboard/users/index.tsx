// libraries
import React, { PureComponent } from 'react';
import moment from 'moment';
import { ThemeProvider } from 'styled-components';

// components
import {
  chartTheme,
  Container,
  GraphsContainer,
  Line,
  GraphCard,
  GraphCardInner,
  GraphCardTitle,
  ControlBar
} from '../';
import TimeControl from '../components/TimeControl';
import IntervalControl from '../components/IntervalControl';
import GenderChart from '../components/GenderChart';
import AgeChart from '../components/AgeChart';

// i18n
import { FormattedMessage } from 'utils/cl-intl';
import messages from '../messages';

interface State {
  interval: 'weeks' | 'months' | 'years';
  intervalIndex: number;
  currentGroupFilter?: string;
}

type Props = {

};

export default class UsersDashboard extends PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      interval: 'months',
      intervalIndex: 0,
      currentGroupFilter: undefined
    };
  }

  changeInterval = (interval: 'weeks' | 'months' | 'years') => {
    this.setState({ interval, intervalIndex: 0 });
  }

  changeIntervalIndex = (intervalIndex: number) => {
    this.setState({ intervalIndex });
  }

  handleOnGroupFilter = (filter) => {
    // To be implemented
    this.setState({ currentGroupFilter: filter });
  }

  render() {
    const { interval, intervalIndex, currentGroupFilter } = this.state;
    const startAtMoment = moment().startOf(interval).add(intervalIndex, interval);
    const endAtMoment = moment(startAtMoment).add(1, interval);
    const startAt = startAtMoment.toISOString();
    const endAt = endAtMoment.toISOString();

    return (
      <Container>
        <ControlBar>
          <TimeControl
            value={intervalIndex}
            interval={interval}
            onChange={this.changeIntervalIndex}
            currentTime={startAtMoment}
          />
          <IntervalControl
            value={interval}
            onChange={this.changeInterval}
          />
        </ControlBar>
        <ThemeProvider theme={chartTheme}>
          <GraphsContainer>
            <Line>
              <GraphCard className="first halfWidth">
                <GraphCardInner>
                  <GraphCardTitle>
                    <FormattedMessage {...messages.usersByGenderTitle} />
                  </GraphCardTitle>
                  <GenderChart startAt={startAt} endAt={endAt} currentGroupFilter={currentGroupFilter} />
                </GraphCardInner>
              </GraphCard>
              <GraphCard className="halfWidth">
                <GraphCardInner>
                  <GraphCardTitle>
                    <FormattedMessage {...messages.usersByAgeTitle} />
                  </GraphCardTitle>
                  <AgeChart startAt={startAt} endAt={endAt} currentGroupFilter={currentGroupFilter} />
                </GraphCardInner>
              </GraphCard>
            </Line>
          </GraphsContainer>
        </ThemeProvider>
      </Container>
    );
  }
}
