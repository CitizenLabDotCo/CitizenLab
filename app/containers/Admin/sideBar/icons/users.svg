<svg xmlns="http://www.w3.org/2000/svg" width="21" height="22" viewBox="0 0 21 22">
    <g fill="none" fillRule="evenodd" stroke="#FFF" stroke-linecap="round" stroke-linejoin="round" stroke-width="2" opacity=".6" transform="translate(1 1)">
        <circle cx="9.091" cy="10" r="9.091"/>
        <circle cx="9.091" cy="8.977" r="2.537"/>
        <path d="M9.09 12.198a5.074 5.074 0 0 0-5.065 5.346 9.09 9.09 0 0 0 5.066 1.547 9.09 9.09 0 0 0 5.06-1.548c.007-.09.012-.18.014-.27a5.074 5.074 0 0 0-5.074-5.075z"/>
    </g>
</svg>
