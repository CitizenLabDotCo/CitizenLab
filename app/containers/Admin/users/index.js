import React from 'react';
import PropTypes from 'prop-types';
import HelmetIntl from 'components/HelmetIntl';

// components
// import { FormattedMessage } from 'react-intl';
// import WatchSagas from 'utils/containers/watchSagas';
// import sagas from 'resources/users/sagas';

// store
// import { preprocess } from 'utils';

// messages
import messages from './messages';

class UserDashboard extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function

  render() {
    return (
      <div>
        <HelmetIntl
          title={messages.helmetTitle}
          description={messages.helmetDescription}
        />
        {this.props.children}
      </div>

    );
  }
}


UserDashboard.propTypes = {
  children: PropTypes.element,
};

export default UserDashboard;
