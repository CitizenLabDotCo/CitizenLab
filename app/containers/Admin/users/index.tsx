import React, { SFC } from 'react';

// components
import HelmetIntl from 'components/HelmetIntl';
import TabbedResource from 'components/admin/TabbedResource';

// resources
import GetLocation, { GetLocationChildProps } from 'resources/GetLocation';

// i18n
import messages from './messages';
import { InjectedIntlProps } from 'react-intl';
import { injectIntl } from 'utils/cl-intl';

interface InputProps {}

interface DataProps {
  location: GetLocationChildProps;
}

interface Props extends InputProps, DataProps {}

const UsersPage: SFC<Props & InjectedIntlProps> = ({ location, children, intl }) => {
  const { formatMessage } = intl;

  const tabs = [
    { label: formatMessage(messages.tabRegisteredUsers), url: '/admin/users/registered' },
    { label: formatMessage(messages.tabInviteByEmail), url: '/admin/users/invitations' },
  ];

  const resource = {
    title: formatMessage(messages.viewPublicResource)
  };

  if (!location) return null;

  return (
    <>
      <TabbedResource
        resource={resource}
        messages={messages}
        tabs={tabs}
        location={location}
      >
        <HelmetIntl
          title={messages.helmetTitle}
          description={messages.helmetDescription}
        />
        {children}
      </TabbedResource>
    </>
  );
};

const UsersPageWithIntl = injectIntl(UsersPage);

export default (inputProps: InputProps) => (
  <GetLocation>
    {location => <UsersPageWithIntl {...inputProps} location={location} />}
  </GetLocation>
);
