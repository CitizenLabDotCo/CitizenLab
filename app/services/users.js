import { API_PATH } from 'containers/App/constants';
import Streams from 'utils/streams';

function getEndpoint() {
  return `${API_PATH}/users`;
}

export function observeUsers() {
  const apiEndpoint = getEndpoint();
  return Streams.create(apiEndpoint);
}
