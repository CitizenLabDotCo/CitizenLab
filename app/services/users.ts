import { API_PATH } from 'containers/App/constants';
import streams, { IStreamParams } from 'utils/streams';
import { API, Multiloc, Locale } from 'typings';

const apiEndpoint = `${API_PATH}/users`;

export interface IRole {
  type: string;
  [key: string]: string;
}

export interface IUserData {
  id: string;
  type: string;
  attributes: {
    first_name: string;
    last_name: string | null;
    slug: string;
    locale: Locale;
    avatar: API.ImageSizes,
    roles?: IRole[],
    bio_multiloc: Multiloc,
    created_at: string;
    updated_at: string;
    email?: string;
    gender?: 'male' | 'female' | 'unspecified';
    birthyear?: number;
    domicile?: string;
    education?: string;
    unread_notifications?: number;
    custom_field_values: object;
  };
}

export interface IUserLinks {
  self: string;
  first: string;
  prev: string;
  next: string;
  last: string;
}

export interface IUsers {
  data: IUserData[];
  links: IUserLinks;
}

export interface IUser {
  data: IUserData;
}

export interface IUserUpdate {
  first_name?: string;
  last_name?: string;
  email?: string;
  password?: string;
  locale?: string;
  avatar?: string;
  roles?: any[];
  birthyear?: number;
  gender?: string;
  domicile?: string;
  education?: string;
  bio_multiloc?: Multiloc;
  custom_field_values?: object;
}

export function usersStream(streamParams: IStreamParams | null = null) {
  return streams.get<IUsers>({ apiEndpoint, ...streamParams });
}

export function userByIdStream(userId: string) {
  return streams.get<IUser>({ apiEndpoint: `${apiEndpoint}/${userId}` });
}

export function userBySlugStream(userSlug: string) {
  return streams.get<IUser>({ apiEndpoint: `${apiEndpoint}/by_slug/${userSlug}` });
}

export async function updateUser(userId: string, object: IUserUpdate) {
  return streams.update<IUser>(`${apiEndpoint}/${userId}`, userId, { user: object });
}

export async function deleteUser(userId: string) {
  return streams.delete(`${apiEndpoint}/${userId}`, userId);
}

export function mapUserToDiff(user: IUserData): IUserUpdate {
  return {
    first_name: user.attributes.first_name || undefined,
    last_name: user.attributes.last_name || undefined,
    email: user.attributes.email || undefined,
    locale: user.attributes.locale || undefined,
    birthyear: user.attributes.birthyear || undefined,
    gender: user.attributes.gender || undefined,
    domicile: user.attributes.domicile || undefined,
    education: user.attributes.education || undefined,
    bio_multiloc: user.attributes.bio_multiloc || undefined,
  };
}
