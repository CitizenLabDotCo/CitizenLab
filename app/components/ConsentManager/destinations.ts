import { isFeatureActive } from 'components/FeatureFlag';
import { ITenantData } from 'services/tenant';
import { IUserData } from 'services/users';

export interface IDestinationMap {
  google_analytics: 'google_analytics';
}

export type IDestination = IDestinationMap[keyof IDestinationMap];
export const CATEGORIES = ['analytics', 'advertising', 'functional'] as const;
export type TCategory = typeof CATEGORIES[number];

const destinationConfigs: IDestinationConfig[] = [];

export interface IDestinationConfig {
  key: IDestination;
  category: TCategory;
  feature_flag?: string;
  hasPermission?: (user?: IUserData) => boolean;
}

export const getDestinationConfigs = () => {
  return destinationConfigs;
};

export const allCategories = () => {
  return CATEGORIES;
};

export const registerDestination = (destinationConfig: IDestinationConfig) => {
  destinationConfigs.push(destinationConfig);
};

export const isDestinationActive = (
  config: IDestinationConfig,
  tenant: ITenantData,
  user?: IUserData | null
): boolean => {
  if (config?.feature_flag && !isFeatureActive(config.feature_flag, tenant)) {
    return false;
  }

  if (config?.hasPermission && !config?.hasPermission(user || undefined)) {
    return false;
  }

  return true;
};
