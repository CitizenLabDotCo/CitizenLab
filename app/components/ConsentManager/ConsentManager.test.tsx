// libraries
import React from 'react';
import { shallow } from 'enzyme';

// component to test
import { ConsentManager } from './';

// mock depencies
jest.mock('services/tenant');
jest.mock('resources/GetTenant', () => 'GetTenant');
jest.mock('./Container', () => 'Container');

// by default, no cookie

jest.mock('./consent', () => ({
  getConsent: jest.fn(() => null),
  setConsent: jest.fn(),
}));

import * as consent from './consent';

import { makeUser } from 'services/__mocks__/users';
import { DESTINATIONS } from './destinations';

// object will all destinations as keys, true as values, mimicks the savedvales for a user that accepted all
const featureFagAllDisabled = DESTINATIONS.map((destination) => [
  destination,
  false,
]).reduce((a, [k, v]) => ({ ...a, [k]: v }), {});
const featureFagAllEnabled = DESTINATIONS.map((destination) => [
  destination,
  true,
]).reduce((a, [k, v]) => ({ ...a, [k]: v }), {});

const admin = makeUser({
  roles: [{ type: 'admin' }],
}).data;

describe('<ConsentManager />', () => {
  describe('boundaries: ', () => {
    it('renders null when tenant is null', () => {
      const wrapper = shallow(<ConsentManager authUser={null} tenant={null} />);
      expect(wrapper.isEmptyRender()).toBe(true);
    });
    it('renders null when tenant is Error', () => {
      const wrapper = shallow(
        <ConsentManager authUser={null} tenant={new Error()} />
      );
      expect(wrapper.isEmptyRender()).toBe(true);
    });
    it('renders with a valid tenant', () => {
      const wrapper = shallow(<ConsentManager authUser={null} tenant={{}} />);
      expect(wrapper.isEmptyRender()).toBe(false);
    });
  });

  describe('parses tenant setting and user to show active destinations in categories', () => {
    describe('unsingned user', () => {
      it('acts properly when all enabled', () => {
        const wrapper = shallow(
          <ConsentManager
            authUser={null}
            tenant={{}}
            {...featureFagAllEnabled}
          />
        );
        const categorizedDestinations = wrapper.find('Container').props()
          .categorizedDestinations;
        expect(categorizedDestinations).toMatchSnapshot();
      });
      it('acts properly when all disabled', () => {
        const wrapper = shallow(
          <ConsentManager
            authUser={null}
            tenant={{}}
            {...featureFagAllDisabled}
          />
        );
        const categorizedDestinations = wrapper.find('Container').props()
          .categorizedDestinations;
        expect(categorizedDestinations).toMatchSnapshot();
      });
    });

    describe('admin user', () => {
      it('acts properly when all enabled', () => {
        const wrapper = shallow(
          <ConsentManager
            authUser={admin}
            tenant={{}}
            {...featureFagAllEnabled}
          />
        );
        const categorizedDestinations = wrapper.find('Container').props()
          .categorizedDestinations;
        expect(categorizedDestinations).toMatchSnapshot();
      });
      it('acts properly when all disabled', () => {
        const wrapper = shallow(
          <ConsentManager
            authUser={admin}
            tenant={{}}
            {...featureFagAllDisabled}
          />
        );
        const categorizedDestinations = wrapper.find('Container').props()
          .categorizedDestinations;
        expect(categorizedDestinations).toMatchSnapshot();
      });
      it('acts properly when only satismeter disabled', () => {
        const wrapper = shallow(
          <ConsentManager
            authUser={admin}
            tenant={{}}
            {...featureFagAllEnabled}
            satismeter={false}
          />
        );
        const categorizedDestinations = wrapper.find('Container').props()
          .categorizedDestinations;
        expect(categorizedDestinations).toMatchSnapshot();
      });
    });
    describe('super admin user', () => {
      const superAdmin = makeUser({
        roles: [{ type: 'admin' }],
        highest_role: 'super_admin',
      }).data;
      it('acts properly when all enabled', () => {
        const wrapper = shallow(
          <ConsentManager
            authUser={superAdmin}
            tenant={{}}
            {...featureFagAllEnabled}
          />
        );
        const categorizedDestinations = wrapper.find('Container').props()
          .categorizedDestinations;
        expect(categorizedDestinations).toMatchSnapshot();
      });
      it('acts properly when all disabled', () => {
        const wrapper = shallow(
          <ConsentManager
            authUser={superAdmin}
            tenant={{}}
            {...featureFagAllDisabled}
          />
        );
        const categorizedDestinations = wrapper.find('Container').props()
          .categorizedDestinations;
        expect(categorizedDestinations).toMatchSnapshot();
      });
    });
  });

  describe('passes down preferences and preferences handling', () => {
    describe('no cookie previously', () => {
      describe('all destinations', () => {
        it('initializes preferences object correctly', () => {
          const wrapper = shallow(
            <ConsentManager
              authUser={null}
              tenant={{}}
              {...featureFagAllEnabled}
            />
          );
          const preferences = wrapper.find('Container').props().preferences;
          expect(preferences).toEqual({
            analytics: undefined,
            advertising: undefined,
            functional: undefined,
          });
        });
        it('changes it as required', () => {
          const wrapper = shallow(
            <ConsentManager
              authUser={null}
              tenant={{}}
              {...featureFagAllEnabled}
            />
          );

          const setPreferences = wrapper.find('Container').props()
            .setPreferences;

          setPreferences({ analytics: false });

          const preferences = wrapper.find('Container').props().preferences;
          expect(preferences).toEqual({
            analytics: false,
            advertising: undefined,
            functional: undefined,
          });
        });
        it('consent is required', () => {
          const wrapper = shallow(
            <ConsentManager
              authUser={null}
              tenant={{}}
              {...featureFagAllEnabled}
            />
          );

          const isConsentRequired = wrapper.find('Container').props()
            .isConsentRequired;

          expect(isConsentRequired).toBe(true);
        });
      });
      describe('no destinations', () => {
        it('initializes preferences object correctly', () => {
          const wrapper = shallow(
            <ConsentManager
              authUser={null}
              tenant={{}}
              {...featureFagAllDisabled}
            />
          );
          const preferences = wrapper.find('Container').props().preferences;
          expect(preferences).toEqual({
            analytics: undefined,
            advertising: undefined,
            functional: undefined,
          });
        });
      });
    });
    describe('previously accepted, no new destinations', () => {
      it('initializes preferences object correctly', () => {
        // COOkIE mock
        // object will all destinations as keys, true as values, mimicks the savedvales for a user that accepted all
        const mock_savedChoices = DESTINATIONS.map((destination) => [
          destination,
          true,
        ]).reduce((a, [k, v]) => ({ ...a, [k]: v }), {});
        const spy = jest
          .spyOn(consent, 'getConsent')
          .mockImplementation(() => ({
            analytics: true,
            advertising: true,
            functional: true,
            savedChoices: mock_savedChoices,
          }));
        const wrapper = shallow(
          <ConsentManager
            authUser={null}
            tenant={{}}
            {...featureFagAllEnabled}
          />
        );
        const preferences = wrapper.find('Container').props().preferences;
        expect(preferences).toEqual({
          analytics: true,
          advertising: true,
          functional: true,
        });
        spy.mockRestore();
      });

      it('consent is not required', () => {
        // COOkIE mock
        // object will all destinations as keys, true as values, mimicks the savedvales for a user that accepted all
        const mock_savedChoices = DESTINATIONS.map((destination) => [
          destination,
          true,
        ]).reduce((a, [k, v]) => ({ ...a, [k]: v }), {});
        const spy = jest
          .spyOn(consent, 'getConsent')
          .mockImplementation(() => ({
            analytics: true,
            advertising: true,
            functional: true,
            savedChoices: mock_savedChoices,
          }));
        const wrapper = shallow(
          <ConsentManager
            authUser={null}
            tenant={{}}
            {...featureFagAllEnabled}
          />
        );

        const isConsentRequired = wrapper.find('Container').props()
          .isConsentRequired;

        expect(isConsentRequired).toBe(false);

        spy.mockRestore();
      });
    });
    describe('previously accepted, new destinations', () => {
      const mock_savedChoices = DESTINATIONS.filter((e) => e !== 'intercom')
        .map((destination) => [destination, true])
        .reduce((a, [k, v]) => ({ ...a, [k]: v }), {});
      it('initializes preferences object correctly', () => {
        // COOkIE mock
        const spy = jest
          .spyOn(consent, 'getConsent')
          .mockImplementation(() => ({
            analytics: true,
            advertising: true,
            functional: true,
            savedChoices: mock_savedChoices,
          }));

        const wrapper = shallow(
          <ConsentManager
            authUser={admin}
            tenant={{}}
            {...featureFagAllEnabled}
          />
        );
        const preferences = wrapper.find('Container').props().preferences;
        expect(preferences).toEqual({
          analytics: true,
          advertising: true,
          functional: undefined,
        });
        spy.mockRestore();
      });

      it('consent is required', () => {
        const spy = jest
          .spyOn(consent, 'getConsent')
          .mockImplementation(() => ({
            analytics: true,
            advertising: true,
            functional: true,
            savedChoices: mock_savedChoices,
          }));
        const wrapper = shallow(
          <ConsentManager
            authUser={admin}
            tenant={{}}
            {...featureFagAllEnabled}
          />
        );

        const isConsentRequired = wrapper.find('Container').props()
          .isConsentRequired;

        expect(isConsentRequired).toBe(true);

        spy.mockRestore();
      });
    });
    describe('previously refused', () => {
      it('initializes preferences object correctly', () => {
        const mock_savedChoices = DESTINATIONS.map((destination) => [
          destination,
          false,
        ]).reduce((a, [k, v]) => ({ ...a, [k]: v }), {});
        const spy = jest
          .spyOn(consent, 'getConsent')
          .mockImplementation(() => ({
            analytics: false,
            advertising: false,
            functional: false,
            savedChoices: mock_savedChoices,
          }));
        const wrapper = shallow(
          <ConsentManager
            authUser={null}
            tenant={{}}
            {...featureFagAllEnabled}
          />
        );
        expect(mock_savedChoices).toMatchSnapshot();
        const preferences = wrapper.find('Container').props().preferences;
        expect(preferences).toEqual({
          analytics: false,
          advertising: false,
          functional: false,
        });
        spy.mockRestore();
      });
    });
  });
});
