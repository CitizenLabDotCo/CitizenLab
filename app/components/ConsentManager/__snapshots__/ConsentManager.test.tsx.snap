// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConsentManager /> parses tenant setting and user to show active destinations in categories admin user acts properly when all disabled 1`] = `
Object {
  "advertising": Array [],
  "analytics": Array [],
  "functional": Array [],
}
`;

exports[`<ConsentManager /> parses tenant setting and user to show active destinations in categories admin user acts properly when all enabled 1`] = `
Object {
  "advertising": Array [],
  "analytics": Array [
    "google_analytics",
    "satismeter",
  ],
  "functional": Array [
    "intercom",
  ],
}
`;

exports[`<ConsentManager /> parses tenant setting and user to show active destinations in categories admin user acts properly when only satismeter disabled 1`] = `
Object {
  "advertising": Array [],
  "analytics": Array [
    "google_analytics",
  ],
  "functional": Array [
    "intercom",
  ],
}
`;

exports[`<ConsentManager /> parses tenant setting and user to show active destinations in categories super admin user acts properly when all disabled 1`] = `
Object {
  "advertising": Array [],
  "analytics": Array [],
  "functional": Array [],
}
`;

exports[`<ConsentManager /> parses tenant setting and user to show active destinations in categories super admin user acts properly when all enabled 1`] = `
Object {
  "advertising": Array [],
  "analytics": Array [
    "google_analytics",
  ],
  "functional": Array [],
}
`;

exports[`<ConsentManager /> parses tenant setting and user to show active destinations in categories unsingned user acts properly when all disabled 1`] = `
Object {
  "advertising": Array [],
  "analytics": Array [],
  "functional": Array [],
}
`;

exports[`<ConsentManager /> parses tenant setting and user to show active destinations in categories unsingned user acts properly when all enabled 1`] = `
Object {
  "advertising": Array [],
  "analytics": Array [
    "google_analytics",
  ],
  "functional": Array [],
}
`;

exports[`<ConsentManager /> passes down preferences and preferences handling previously refused initializes preferences object correctly 1`] = `
Object {
  "google_analytics": false,
  "intercom": false,
  "satismeter": false,
}
`;
